src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -150,20 +155,19 @@ public static string CreateWorkItemDescription(this SarifLog log, SarifWorkItemC
 
             IEnumerable<Result> results = log?.Runs?[0]?.Results.Where(r => r.ShouldBeFiled());
             Uri runRepositoryUri = log?.Runs.FirstOrDefault()?.VersionControlProvenance?.FirstOrDefault().RepositoryUri;
-            Uri detectionLocationUri = !string.IsNullOrEmpty(runRepositoryUri?.OriginalString) ?
-                                       runRepositoryUri :
-                                       results?.FirstOrDefault().Locations?[0].PhysicalLocation?.ArtifactLocation?.Uri;
+            Uri detectionLocationUri = !string.IsNullOrEmpty(runRepositoryUri?.OriginalString)
+                ? runRepositoryUri
+                : results?.FirstOrDefault()?.Locations?[0].PhysicalLocation?.ArtifactLocation?.Uri;
 
             string detectionLocation = (detectionLocationUri?.IsAbsoluteUri == true && detectionLocationUri?.Scheme == "https")
                 ? context.CreateLinkText(detectionLocationUri.OriginalString, detectionLocationUri?.OriginalString)
                 : detectionLocationUri?.OriginalString;
 
-            int locCount = results == null ? 0 :
-                           results
-                           .Where(r => r.Locations != null)
-                           .SelectMany(r => r.Locations)
-                           .Where(l => l.PhysicalLocation != null && l.PhysicalLocation.ArtifactLocation != null && l.PhysicalLocation.ArtifactLocation.Uri != null)
-                           .Count();
+            int locCount = results == null
+                ? 0
+                : results.Where(r => r.Locations != null)
+                         .SelectMany(r => r.Locations)
+                         .Count(l => l.PhysicalLocation?.ArtifactLocation?.Uri != null);
```

Comment:
my personal opinion is this code isn't super readable and i avoid linq statements that potentially entail multiple iterations over a collection (in a pathological case).